server/db.go

```diff
@@ -415,192 +335,111 @@ func (c *Server) fetchIssuerByType(ctx context.Context, issuerType string) (*Iss
-		}
-
-		var fetchIssuerKeys = []issuerKeys{}
-		err = c.db.Select(
-			&fetchIssuerKeys,
-			`SELECT *
-			FROM v3_issuer_keys where issuer_id=$1 and 
-			(
-				(select version from v3_issuers where issuer_id=$1)<=2
-				or end_at > now()
-			)
-			ORDER BY end_at ASC NULLS FIRST, start_at ASC`,
-			convertedIssuer.ID,
-		)
-		if err != nil {
-			if !isPostgresNotFoundError(err) {
-				c.Logger.Error("Failed to extract issuer keys from DB")
-				temporary = true
-			}
-			return nil, utils.ProcessingErrorFromError(err, temporary)
-		}
-
-		for _, v := range fetchIssuerKeys {
-			k, err := c.convertDBIssuerKeys(v)
-			if err != nil {
-				c.Logger.Error("Failed to convert issuer keys from DB")
-				return nil, utils.ProcessingErrorFromError(err, temporary)
-			}
-			convertedIssuer.Keys = append(convertedIssuer.Keys, *k)
-		}
-
-		issuers = append(issuers, *convertedIssuer)
+	issuersWithKey, err := c.fetchIssuerKeys(fetchedIssuers)
+	if err != nil {
+		return nil, err
 	}
 
 	if c.caches != nil {
-		c.caches["issuers"].SetDefault(issuerType, issuers)
+		c.caches["issuers"].SetDefault(issuerType, issuersWithKey)
 	}
 
-	return issuers, nil
+	return issuersWithKey, nil
 }
 
-// FetchAllIssuers fetches all issuers from a cache or a database, saving them in the cache
-// if it has to query the database.
-func (c *Server) FetchAllIssuers() (*[]Issuer, error) {
-	if cached := retrieveFromCache(c.caches, "issuers", "all"); cached != nil {
-		if issuers, ok := cached.(*[]Issuer); ok {
-			return issuers, nil
-		}
-	}
-
-	var (
-		err       error
-		temporary = false
-	)
-
-	fetchedIssuers := []issuer{}
-	err = c.db.Select(
-		&fetchedIssuers,
-		`SELECT *
-		FROM v3_issuers
-		ORDER BY expires_at DESC NULLS LAST, created_at DESC`)
-	if err != nil {
-		c.Logger.Error("Failed to extract issuers from DB")
-		if !isPostgresNotFoundError(err) {
-			temporary = true
-		} else {
-			panic("Postgres encountered temporary error")
-		}
-		return nil, utils.ProcessingErrorFromError(err, temporary)
-	}
-
-	issuers := []Issuer{}
+func (c *Server) fetchIssuerKeys(fetchedIssuers []issuer) ([]Issuer, error) {
+	var issuers []Issuer
 	for _, fetchedIssuer := range fetchedIssuers {
 		convertedIssuer := c.convertDBIssuer(fetchedIssuer)
-
+		// get the keys for the Issuer
 		if convertedIssuer.Keys == nil {
 			convertedIssuer.Keys = []IssuerKeys{}
 		}
 
-		var fetchIssuerKeys = []issuerKeys{}
-		err = c.db.Select(
-			&fetchIssuerKeys,
+		var keys []issuerKeys
+		lteVersionTwo := fetchedIssuer.Version <= 2
+		// TODO: There are tests relying on the lte V2 check to do a full table scan?
+		err := c.db.Select(
+			&keys,
 			`SELECT *
-			FROM v3_issuer_keys where issuer_id=$1 and
-			(
-				(select version from v3_issuers where issuer_id=$1)<=2
-				or end_at > now()
-			)
-			ORDER BY end_at ASC NULLS FIRST, start_at ASC`,
-			convertedIssuer.ID,
+			FROM v3_issuer_keys 
+			WHERE issuer_id=$1
+			  AND ($2 OR (end_at > now() OR end_at is null))
```

Comment:
After a lot of messing around with this, it appears that there are some tests that fully depend on `version <=2 OR end at is in the future` so basically this statement becomes:
```
SELECT * from v3_issuer_keys where issuer_id = 'id' AND true;
```

I have now changed the SQL to properly order the data, wondering if it should be left there, or more should be done